src/hooks/use-cart.ts

```diff
@@ -22,28 +25,54 @@ export const useCart = () => {
 
     const updateItemQuantity = ({ id, quantity }: { id: string; quantity: number }) => {
         setUpdatingCartItems((prev) => [...prev, id]);
-        triggerUpdateItemQuantity({ id, quantity }).finally(() => {
-            setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
-        });
+        triggerUpdateItemQuantity({ id, quantity })
+            .catch((error) => {
+                runToast({
+                    type: ToastType.Error,
+                    message: `Failed to update the item quantity. ${getErrorMessage(error)}`,
+                });
+                throw error;
```

Comment:
`throw error;` - why?

Reply:
_use-cart_
`const addToCart = (args) => triggerAddToCart(args).catch((error) => runToast(...));`

usage:
`addToCart(args).then(() => cartOpener.setIsOpen(true));`

In this case `then` executes. I should have thrown error in `catch`.

But never mind, I'll move toast calls out of `useCart` hook.